app/Http/Controllers/UserController.php

```diff
@@ -161,44 +161,8 @@ public static function getProfilePage($username): ?array {
         }
 
 
-        $twitterUrl  = "";
-        $mastodonUrl = "";
-
-
-        if ($user->socialProfile != null) {
-            try {
-                $mastodonServer = MastodonServer::where('id', $user->socialProfile->mastodon_server)->first();
-                if ($mastodonServer != null) {
-                    $mastodonDomain      = $mastodonServer->domain;
-                    $mastodonAccountInfo = Mastodon::domain($mastodonDomain)
-                                                   ->token($user->socialProfile->mastodon_token)
-                                                   ->get("/accounts/" . $user->socialProfile->mastodon_id);
-                    $mastodonUrl         = $mastodonAccountInfo["url"];
-                }
-            } catch (Exception) {
-                // The connection might be broken, or the instance is down, or $user has removed the api rights
-                // but has not told us yet.
-            }
-        }
-
-        if ($user->socialProfile != null) {
-            if (!empty($user->socialProfile->twitter_token) && !empty($user->socialProfile->twitter_tokenSecret)) {
-                try {
-                    $connection = new TwitterOAuth(
-                        config('trwl.twitter_id'),
-                        config('trwl.twitter_secret'),
-                        $user->socialProfile->twitter_token,
-                        $user->socialProfile->twitter_tokenSecret
-                    );
-
-                    $getInfo    = $connection->get('users/show', ['user_id' => $user->socialProfile->twitter_id]);
-                    $twitterUrl = "https://twitter.com/" . $getInfo->screen_name;
-                } catch (Exception $e) {
-                    // The big whale time or $user has removed the api rights but has not told us yet.
-                }
-            }
-        }
-
+        $twitterUrl  = $user->twitterUrl;
+        $mastodonUrl = $user->mastodonUrl;
```

Comment:
Good Idea! However, I would now still remove these two lines from the controller and instead simply access the user variable in the blade. This saves another 4 lines.

Reply:
I even reduced the amount of variable declarations by just moving the attribute getters to the array.